libsycl/CMakeLists.txt

```diff
@@ -0,0 +1,158 @@
+#===============================================================================
+# Setup Project
+#===============================================================================
+cmake_minimum_required(VERSION 3.20.0)
+
+set(LLVM_SUBPROJECT_TITLE "libsycl")
+
+set(LIBSYCL_SOURCE_DIR  ${CMAKE_CURRENT_SOURCE_DIR})
+set(LIBSYCL_BINARY_DIR  ${CMAKE_CURRENT_BINARY_DIR})
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+#===============================================================================
+# Setup CMake Options
+#===============================================================================
+
+option(LIBSYCL_ENABLE_WERROR "Treat all warnings as errors in the libsycl project" OFF)
+
+#===============================================================================
+# Configure System
+#===============================================================================
+
+set(LIBSYCL_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}" CACHE STRING
+    "Path where libsycl headers should be installed.")
+set(LIBSYCL_INSTALL_LIBRARY_DIR "${CMAKE_INSTALL_BINDIR}" CACHE STRING
```

Comment:
It seems this variables is being overwritten on line 38 or 47 below. If so, this does not make sense to have it defined here.

Besides, On Linux `CMAKE_INSTALL_BINDIR` is typically `/usr/bin` if project is installed system wide. This location is no place for .so libraries - that's location for executables, applications and scripts. Dynamic libraries on Linux should effectively go to `CMAKE_INSTALL_LIBDIR`.

Reply:
removed